js/src/common/Session.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * The `Session` class defines the current user session. It stores a reference
+ * to the current authenticated user, and provides methods to log in/out.
+ */
+import User from './models/User';
+import { RequestOptions } from './Application';
+
+export type LoginParams = {
+  /**
+   * The username/email
+   */
+  identification: string;
+
+  /**
+   * Password
+   */
+  password: string;
+};
+
+export default class Session {
+  /**
+   * The current authenticated user.
+   */
+  public user: User | null;
+
+  /**
+   * The CSRF token.
+   */
+  public csrfToken: string | null;
+
+  constructor(user: User | null, csrfToken: string | null) {
+    this.user = user;
+    this.csrfToken = csrfToken;
+  }
+
+  /**
+   * Attempt to log in a user.
+   */
+  public login(body: LoginParams, options: Omit<RequestOptions, 'url'> = {}) {
+    return app.request(
+      Object.assign(
+        {
+          method: 'POST',
+          url: `${app.forum.attribute('baseUrl')}/login`,
+          body,
+        },
+        options
+      )
```

Comment:
This is more in-line with our new TS, I think

```suggestion
      {
        method: 'POST',
        url: `${app.forum.attribute('baseUrl')}/login`,
        body,
        ...options,
      },
```